### ISSUE TITLE
1817: a company redeemed shares for 0$

### ISSUE THREAD

**Issue by PedroRCSequeira** [tobymao/18xx#2682]
in this game, just after this move, PSNR got one more loans, went into acquisition and redeemed 100% for 0$

https://18xx.games/game/15967?action=559

"PSNR buys a 100% share of PSNR from the market for $0"

**Comment by ventusignis** [comment]
It looks like the corporation simply shouldn't have been allowed to redeem the shares

```
5.6
Acting on a Company's Behalf
If a player decides to act on behalf of a company instead
of taking action himself, he must perform one or both of
the following actions in this order for any one company
for which he is president and whose share price is not in
the red liquidation or gray acquisition areas of the stock
market:
1)
Take   one   or   more   loans   up   to   the   maximum
allowed for the company,  as described in [ 1.2.5 ].
(Reminder:   interest   is   due   each   operating   round,
but not during the stock round.)
2)
Purchase one or more shares of the company’s own
stock   from   the   open   market,   up   to   the   amount
available,   by   paying   the   bank   for   the   stock
purchase.
```

**Comment by jenf** [comment]
Was fixed by #2768
